test/unit/spec/LoginRouter_spec.js

```diff
@@ -581,7 +580,7 @@ function (Okta, Q, Backbone, SharedUtil, CryptoUtil, CookieUtil, Logger, OktaAut
         expect(form.isPrimaryAuth()).toBe(true);
         // Respond with MFA_REQUIRED
         // Verify is immediately called, so respond with MFA_CHALLENGE
-        test.setNextResponse([resMfaRequiredOktaVerify, resMfaChallengePush]);
+        test.setNextResponse([resMfaRequiredOktaVerifyAutoPushTrue, resMfaChallengePush]);
```

Comment:
following my other comment, here instead of using ```resMfaRequiredOktaVerifyAutoPushTrue```, just call a function like ```getAutoPushResponse(resMfaRequiredOktaVerify, true)``` that creates the response you need
```
function getAutoPushResponse(response, autoPushEnabled) {
  // clone the response and change the value you need
  var autoPushResponse = _.clone(response);
  autoPushResponse._embedded.policy.factorsPolicyInfo.autoPush = autoPushEnabled;
  return autoPushResponse;
}
```

Reply:
I tried that first, but the autoPushEnabled property is copied by reference when using _.clone. So subsequent tests fail.. If creating another xhr file is not an option, would it make sense to set this property for the test and reset it at the end?

Reply:
What about something like [deepClone](https://github.com/okta/okta-signin-widget/blob/653ceac778903b8189003802f597b9ba9855a552/test/unit/spec/PasswordExpired_spec.js#L30)?

Reply:
That should work! Thanks!